include/zephyr/bluetooth/hci_types.h

```diff
@@ -108,17 +108,17 @@ struct bt_hci_acl_hdr {
 #define bt_iso_pkt_flags(h)              ((h) >> 14)
 #define bt_iso_pkt_len_pack(h, f)        (((h) & BIT_MASK(12)) | ((f) << 14))
 
-struct bt_hci_iso_data_hdr {
+struct bt_hci_iso_sdu_hdr {
 	uint16_t sn;
 	uint16_t slen; /* 12 bit len, 2 bit RFU, 2 bit packet status */
 } __packed;
-#define BT_HCI_ISO_DATA_HDR_SIZE	4
+#define BT_HCI_ISO_SDU_HDR_SIZE          4
 
-struct bt_hci_iso_ts_data_hdr {
+struct bt_hci_iso_sdu_ts_hdr {
 	uint32_t ts;
-	struct bt_hci_iso_data_hdr data;
+	struct bt_hci_iso_sdu_hdr sdu;
 } __packed;
-#define BT_HCI_ISO_TS_DATA_HDR_SIZE     8
+#define BT_HCI_ISO_SDU_TS_HDR_SIZE       8
 
```

Comment:
I see what you are trying to do here, and I agree the naming could be improved. 

What I'm worried about here is that these are not headers for SDUs, but SDU fragments. 

If the host sends an SDU in 3 fragments to the controller, this header will appear 3 times. 

Would `bt_hci_iso_sdu_frag_xx` be better?

Reply:
Will it though?
I think those "headers" are really part of the whole-sdu payload. Just like the SDU "header" in L2CAP K-frames, it shouldn't be repeated for every fragment. It's only there at the start of the SDU.

I would however be ok with renaming `struct bt_hci_iso_hdr` to `struct bt_hci_iso_data_hdr`:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/68428270/0d8b8e27-70ac-4d93-a0f5-dfa37214f088)

Reply:
> What I'm worried about here is that these are not headers for SDUs, but SDU fragments.

I disagree here, by the Spec:

```
If PB_Flag equals 0b00 or 0b10, then the Packet_Sequence_Number,
ISO_SDU_Length, and Packet_Status_Flag fields (plus the intermediate RFU
field) shall all be present in the packet and the Time_Stamp field may be
present. If PB_Flag equals 0b01 or 0b11, then none of these fields shall be
included in the packet.
```

Hence, `struct bt_hci_iso_sdu_hdr` is once per SDU (first fragment or complete SDU) from the upper layer, these fields are not part of next SDU fragment towards the Controller.

Reply:
> I would however be ok with renaming struct bt_hci_iso_hdr to struct bt_hci_iso_data_hdr

This I can do, and will be closer to the Spec wording `HCI ISO Data packet header`.

Reply:
> > What I'm worried about here is that these are not headers for SDUs, but SDU fragments.
> 
> I disagree here, by the Spec:
> 
> ```
> If PB_Flag equals 0b00 or 0b10, then the Packet_Sequence_Number,
> ISO_SDU_Length, and Packet_Status_Flag fields (plus the intermediate RFU
> field) shall all be present in the packet and the Time_Stamp field may be
> present. If PB_Flag equals 0b01 or 0b11, then none of these fields shall be
> included in the packet.
> ```
> 
> Hence, `struct bt_hci_iso_sdu_hdr` is once per SDU (first fragment or complete SDU) from the upper layer, these fields are not part of next SDU fragment towards the Controller.

I stand corrected. Thanks! 

In this case, I'm OK with the new naming

Reply:
super confusing as always from the SIG. And the next bit in the same paragraph is cool too "SDU fragments generated over HCI are unrelated to the SDU fragments generated by ISOAL".

Reply:
Indeed. The entire ISO over HCI is a complete mess :(